src/famfs_lib.c

```diff
@@ -236,24 +236,78 @@ famfs_print_uuid(const uuid_le *uuid)
 	printf("%s\n", uuid_str);
 }
 
-#define SYS_UUID_PATH "/sys/devices/virtual/dmi/id/product_uuid"
+#define SYS_UUID_FILE "/opt/famfs/system_uuid"
+#define SYS_UUID_DIR "/opt/famfs"
```

Comment:
I think we will want to use the unit test framework for some of the testing of the uuid file. I think every failure mode in this path could be reproduced there. That would be easier if you do
char *sys_uuid_path = ...
...so it can be overridden easily in unit tests.

Also, would be nice not to have SYS_UUID_DIR and SYS_UUID_FILE contain any redundant content (i.e. /opt/fafs is part of both).

Reply:
Agreed.
Added unit tests for famfs_create_sys_uuid_file( char *sys_uuid_file) and changed the function signature.
Removed redundant content from SYS_UUID_DIR and SYS_UUID_FILE